alfresco-transform-core-aio/alfresco-transform-core-aio-boot/src/test/java/org/alfresco/transformer/AIOControllerImageMagickTest.java

```diff
@@ -0,0 +1,152 @@
+import org.alfresco.transformer.transformers.ImageMagickAdapter;
+import org.alfresco.transformer.transformers.Transformer;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.SpyBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.mock.web.MockMultipartFile;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.test.web.servlet.request.MockHttpServletRequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(AIOController.class)
+@Import(AIOCustomConfig.class)
+public class AIOControllerImageMagickTest extends ImageMagickControllerTestBase 
+{
+    
+    static ImageMagickAdapter adapter;
+    
+    @Autowired
+    AllInOneTransformer transformer;
+
+    @SpyBean
+    AIOController controller;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception
+    {
+        adapter = new ImageMagickAdapter();
+    }
+
+    @Before @SuppressWarnings("unchecked")
+    public void before() throws IOException, Exception
+    {
+        adapter = new ImageMagickAdapter();
+        ReflectionTestUtils.setField(commandExecutor, "transformCommand", mockTransformCommand);
+        ReflectionTestUtils.setField(commandExecutor, "checkCommand", mockCheckCommand);
+        ReflectionTestUtils.setField(adapter, "commandExecutor", commandExecutor);
+        //Need to wire in the mocked adpater into the controller...
+        if (ReflectionTestUtils.getField(transformer,"transformerTransformMapping") instanceof Map)
+        {
+            Map<String,Transformer> transformers = (Map<String,Transformer>)ReflectionTestUtils.getField(transformer,"transformerTransformMapping");
+            transformers.replace("imagemagick", adapter);
+            ReflectionTestUtils.setField(transformer, "transformerTransformMapping", transformers);
+        }
+
+        mockTransformCommand("jpg", "png", "image/jpeg", true);
+    }
+
+    @Override
+    protected AbstractTransformerController getController() 
+    {
+        return controller;
+    }
+
+    @Override
+    protected MockHttpServletRequestBuilder mockMvcRequest(String url, MockMultipartFile sourceFile,
+        String... params)
+    {
+        final MockHttpServletRequestBuilder builder = super.mockMvcRequest(url, sourceFile, params)
+            .param("targetMimetype", targetMimetype)
+            .param("sourceMimetype", sourceMimetype);
+
+        return builder;
+    }
+
+    @Test
+    @Override
+    public void noTargetFileTest()
+    {
+        // Ignore the test in super class as the AIO transforms we not be selected .

```

Comment:
Trivial: grammar -> "we not be selected ." -> "was not selected." ? Similar below ...